File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(number)
    @number = number
  end

  def square_of_sum
    number_list.sum.abs2
  end

  def sum_of_squares
    number_list.collect { |i| i.abs2 }.sum
  end

  def difference
    square_of_sum - sum_of_squares
  end

  private

  def number_list
    1..@number
  end
end

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func Version() string {
	return "1"
}

func SquareOfSum(num int) int {
	sum := 0
	for i := 1; i <= num; i++ {
		sum += i
	}
	return sum * sum
}

func SumOfSquares(num int) int {
	sum := 0
	for i := 1; i <= num; i++ {
		sum += i * i
	}
	return sum
}

func Difference(num int) int {
	return SquareOfSum(num) - SumOfSquares(num)
}
